File: 2 SEMESTER/Object Oriented Programming/Projects/PROJECT2_JAVA/src/Organisms/Plants/Plant.java
package Organisms.Plants;

import Game.World;
import Organisms.Organism;

import java.awt.*;
import java.io.Serializable;

public class Plant extends Organism implements Serializable
{
    Plant(String symbol, String name, Color color, int sila, int initative, int speed, World world,int x, int y)
    {
        super(symbol, name, color, sila, initative, speed, world,x,y);
    }
    @Override
    public void Action()
    {
        if(world.getRandomNumber(50)==0)
        {
            if(world.isAnyNearbyPlaceFree(this))
            {
                int newX=0;
                int newY=0;
                while(!world.isPositionWithinBoard(newX,newY) || world.IsPlaceTaken(newX,newY))
                {
                    newX=world.getNewPosition(this.x,1);
                    newY=world.getNewPosition(this.y,1);
                }
                world.addOrganism(this.getChild(newX,newY));
                world.addToLog(this + "rozmnaza sie.");
                world.addToLog("Nowy organizm typu: " + getName() + " pojawi sie na pozycji " + newX + " " + newY);
            }
            else
            {
                world.addToLog(this + " probowal sie rozmnozyc, ale nie bylo na to miejsca. ");
            }
        }
    }

    @Override
    public void Collision(Organism attacker)
    {
        if(attacker.getStrength()>=this.getStrength())
        {
            world.addToLog(attacker + " zjada " + this);
            attacker.setXY(getX(),getY());
            this.kill();
        }
        else
        {
            world.addToLog(attacker + " zjada " + this + " i ginie.");
            this.kill();
            attacker.kill();
        }
    }
}

File: 2 SEMESTER/Object Oriented Programming/Projects/PROJECT3_PYTHON/Organisms/Plants/plant.py
import random

from Organisms.organism import Organism


class Plant(Organism):
    def __init__(self, position_x, position_y, name, symbol, strength, initiative, world, speed, color):
        super().__init__(position_x, position_y, name, symbol, strength, initiative, world, speed, color)

    def action(self):
        if self.world.isbreedingpossible(self):
            i = random.randint(1, 50)
            print(i)
            if i == 1:
                childposition = self.world.findplaceforkid(self)
                self.world.addorganism(self.getchild(childposition))
                self.world.logs.append(self.tostring() + " rozmnaża się. Dziecko pojawi się na " + str(childposition))


    def collision(self, attacker):
        if attacker.strength > self.strength:
            self.world.logs.append(attacker.tostring() + " zjada " + self.tostring())
            attacker.position_x = self.position_x
            attacker.position_y = self.position_y
            self.kill()
            self.planteffect(attacker)
        else:
            self.world.logs.append(attacker.tostring() + " zjada " + self.tostring() + " i ginie.")
            self.kill()
            attacker.kill()

    def planteffect(self, attacker):
        pass
